### ISSUE TITLE
Incorrect group offsets returned

### ISSUE THREAD

**Issue by stoarca** [mscdex/node-pcre#4]
To my knowledge, if an array is returned by PCRE.exec, it must be of even length. I'm encountering some weird behavior when dealing with nested groups. Test case:

```
PCRE.exec('((t))+', 't', 0) // returns [0, 1, 0, 0, 1], which has odd length
```

This seems like a bug. If not, what do the offsets in the array mean?

**Comment by stoarca** [comment]
I think the error is at https://github.com/mscdex/node-pcre/blob/master/src/binding.cc#L545 and
https://github.com/mscdex/node-pcre/blob/master/src/binding.cc#L551

i is being incremented by only 1 instead of 2. I haven't tested to see if this is the actual problem.

**Comment by alexgorbatchev** [comment]
@mscdex could you chip in on this please?

**Comment by mscdex** [comment]
Should be fixed in 0dc7fd9d5.

**Comment by alexgorbatchev** [comment]
Thnx a ton!

**Comment by mscdex** [comment]
Actually fixed in 8549a2ff3. I'll be writing tests soon :-)
